src/v/storage/parser_utils.cc

```diff
@@ -88,34 +88,24 @@ model::record_batch_reader compress_batch_consumer::end_of_stream() {
 }
 
 ss::future<model::record_batch>
-compress_batch(model::compression c, model::record_batch&& b) {
+compress_batch(model::compression c, model::record_batch b) {
```

Comment:
Any reason to not keep the r-value reference?  It does make explicit the need to copy something if we're attempting to compress a record_batch.

Reply:
Some reasons:
* The copy constructor is deleted, so you have to explicitly `std::move(batch)` or `batch.copy()` (unless it's an rvalue-ref)
* It's a coroutine, so parameters shouldn't be references (I can make it normal futurized if really desired)
* The subtle difference between throwing (if it's const&) and returning (if it's rvalue-ref) when it's not compressed is kind of annoying
* It's not used anywhere